src/components/PostDetails.tsx

```diff
@@ -1,117 +1,134 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getComments } from '../api/comments';
+
+type Props = {
+  post: Post | null;
+  setError: (message: string) => void;
+  error: string;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  setError,
+  error,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isNewComment, setIsNewComment] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadApiComments = async (postId: number) => {
+    try {
+      setIsLoading(true);
+      const data = await getComments(postId);
+
+      if (!data.length) {
+        setError('No comments yet');
+      }
+
+      setComments(data);
+    } catch {
+      setError('Unable to get comments');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (post?.id) {
+      loadApiComments(post.id);
+      setError('');
+    }
+  }, [post]);
```

Comment:
I don't think that you need to pass the whole post here 
```suggestion
  }, [post.id]);
```

Reply:
done